### ISSUE TITLE
lsp finder file paths

### ISSUE THREAD

**Issue by matthewsia98** [glepnir/lspsaga.nvim#811]
### Describe the bug

Lspsaga uses the first lsp `root_dir` it finds.
This can be a problem because order of servers can change and for server like Copilot the `root_dir` is always `$HOME`.
Not sure what would be the best way to solve that.

Also, in the second image below, I'm not sure why `user` becomes `ser`.

![image](https://user-images.githubusercontent.com/71731492/216049093-22c93961-214c-4772-9dd4-dd8ff44167f8.png)
![image](https://user-images.githubusercontent.com/71731492/216049150-a55b3e7f-78a0-4f48-9590-9fd88a886f1f.png)

### Steps to reproduce

1. `echo 'local s = "Hello World"
print(s)' > ~/.testing/nvim/lua/user/config.lua
`
2. `nvim -u minimal.lua ~/.testing/nvim/lua/user/config.lua`
3. Put cursor on variable `s`
4. Run `:Lspsaga lsp_finder`

```lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")
-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath })
end
vim.opt.runtimepath:prepend(lazypath)

-- options
vim.opt.termguicolors = true
vim.opt.completeopt = ""
vim.opt.wildmenu = false

-- install plugins
local plugins = {
    "nvim-lua/plenary.nvim",
    "folke/tokyonight.nvim",
    {
        "neovim/nvim-lspconfig",
        dependencies = {
            "williamboman/mason.nvim",
            "williamboman/mason-lspconfig.nvim",
        },
        event = "BufReadPre",
        config = function()
            require("mason").setup()

            -- install mason packages
            local show_ui = true
            local packages = {
                -- "python-lsp-server",
                "lua-language-server",
            }
            for _, package_name in ipairs(packages) do
                local package = require("mason-registry").get_package(package_name)
                if not package:is_installed() then
                    if show_ui then
                        vim.cmd("Mason")
                        show_ui = false
                    end
                    package:install()
                end
            end

            require("mason-lspconfig").setup()

            local cmp_installed, cmp = pcall(require, "cmp_nvim_lsp")
            local capabilities = cmp_installed and cmp.default_capabilities()
                or vim.lsp.protocol.make_client_capabilities()
            require("lspconfig").sumneko_lua.setup({
                on_attach = function(client, bufnr)
                    -- Disable semantic highlighting by server
                    -- sumneko lua highlights comments when toggling
                    local servers_disable_semantic_highlighting = { "sumneko_lua" }
                    if vim.tbl_contains(servers_disable_semantic_highlighting, client.name) then
                        client.server_capabilities.semanticTokensProvider = nil
                    end

                    vim.keymap.set("i", "<C-s>", function()
                        vim.lsp.buf.signature_help()
                    end, { buffer = bufnr })
                end,
                handlers = {},
                capabilities = capabilities,
                flags = { allow_incremental_sync = true, debounce_text_changes = 150 },
                settings = { },
            })
        end,
    },
    {
        "glepnir/lspsaga.nvim",
        event = "BufReadPre",
        config = function()
            require("lspsaga").setup({
                request_timeout = 5000,
            })
        end,
    },
}
require("lazy").setup(plugins, {
    root = root .. "/plugins",
    defaults = { lazy = true },
    ui = { border = "rounded" },
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here

```

### Expected behavior

Don't cut word

### Neovim version (nvim -v)

NVIM v0.9.0-dev-825+g843c9025a

### lspsaga commit

f514660

### Terminal name/version

kitty 0.26.5

**Comment by glepnir** [comment]
finder bug will fix in #810  thanks for report.

**Comment by glepnir** [comment]
root will fix in pr . the letter cut fixed.
